src/npem.c

```diff
@@ -99,13 +87,15 @@ static struct pci_access *get_pci_access()
 static struct pci_dev *get_pci_dev(struct pci_access *pacc, const char *path)
 {
 	unsigned int domain, bus, dev, fn;
-	char *p = strrchr(path, '/');
+	char dpath[PATH_MAX];
+	char *p;
+
 	int ret = 0;
 
-	if (!p)
-		return NULL;
+	snprintf(dpath, PATH_MAX, "%s", path);
+	p = basename(dpath);
 
-	ret += str_toui(&domain, p + 1, &p, 16);
+	ret += str_toui(&domain, p, &p, 16);
 	ret += str_toui(&bus, p + 1, &p, 16);
```

Comment:
Have you been tested if rest of the fields are written correctly when (p+1) is given instead of p?

Reply:
The change does not affect other places because every `str_toui` call modifies `p` pointer and moves it at the end of parsed input part.  The next `p+1`s come from `:` between each part of bdf.
The first `p+1` was caused by `\` from strrchr- it is not a case now with `basename`.
Anyway, I tested it and it works correctly.